### DISCUSSION TITLE
ATS-20 Starter Pack v2.1

### DISCUSSION THREAD

**Discussion by patomoi** [patomoi/ATS-20-Starter-Pack#3] [Q&A]
- fixed a bug in the band names
- added instructions

<hr /><em>This discussion was created from the release <a href='https://github.com/patomoi/ATS-20-Starter-Pack/releases/tag/'>ATS-20 Starter Pack v2.1</a>.</em>

**Comment by RFCanada** [comment]
Hi all,

First I want to say Thank You for the great work on the ATS-20 Starter Pack! 

I just found the work you and others have done to make the ATS-20 better. I love the display's look!

While I'm new to Arduino and have lot's to learn, I have done some minor changes to the latest code that PU2CLR posted on his GitHub page. Small learning increments such as font and playing around with other things to see what happens, all of course keeping a backup of the original code to revert back to normal. I love my ATS-20!

I have found some minor bugs in the ATS-20 SP v2.1 and not sure how to correct them myself being so new to this coding stuff and Arduino.

- The STEP shows 25kHz but when you spin the encoder knob you are in a different step, say 1, 5 or 10. All the bands do this. eg. Say your on the 20m ham band USB, 25kHz STEP is showing but your moving in 1kHz steps as you spin the encoder knob. You can change the step to 1kHz but after you stop, and 10 seconds pass, the STEP shows 25kHz again. Spin the encoder again and the STEP still shows 25kHz displayed but you increment 1kHz on the frequency readout. It's like the default readout is 25kHz.
 
- Is it possible to post some pictures of the Battery Mod? I follow to a point on the instructions but don't want to assume I know what to do for sake of smoking my ATS-20. I own 2 ATS-20's that have two different circuit board designs. One has the nano board on a riser and the other one is a one piece board design. Any ideas on how to do the battery mod on the one with the one board design with "clone" nano on the board?

- The ATS-20 doesn't seem to remember the settings of each band. I set up 3.750 LSB and all the bands now are in LSB as I scroll through them. 14.200 USB is now LSB. Is this because of low memory? I utilised 97% when loading the non-battery mod version. Battery mod version uses 99%. Just an observation.

- Can the user manual instructions explain the function and how to use AVC and SoftMute?

Over-all I love the new design of the display. Not sure what AVC really does or the SoftMute. If I knew more about coding and Arduino I'd try and figure these issues out myself. Again keep up the great work and allowing me to learn a little about SDR/Arduino/coding.


Rob

**Reply by patomoi** [reply]
Hi Rob. 

I'm glad to see that you like the starter pack  proyect. This proyect born of the fusion of the -now outdated- battery mod and band name proyect by Darren and the Ricardo lasts updates to the firmware. 

About the soft mute and avc functions I don't have notice what functions really does, seems that soft mute attenuates the change between frecuencies and avc regulates the loudness of the speaker. 

I have 2 ats 20 too in the same way like you, the batt mod is only possible in the version with a nano in riser. I have no success in detect the points where make the divisor in the non nano version. The batt mod is simple, put a tension divider (2 10Kohm) between positive lead in batt, pin a2 and ground in Arduino. 

+Batt----10K----a2---10K---arduino ground. 

About the save function, the device only save the frecuency , bandwidth and the step, the AGC, volume and mode are global

The 25k step refers to the step of the bfo and not the step of the frecuency change, if  you deactivate the sideband mode you will see that step match with the change on freq and in sideband mode the step is the step in bfo. 

This issue I think is about the patch used to make the sideband mode to work, I have not be able to fix that. 

Would be great some help and suggestions about  improvement about the sketch

In reality, the sketch is very limited in free space right now, although I already have done some minor clean to the code. The solution would be delete some unused bands and deactivate some functions

I was working in a display modification that would let shut off-on  the screen  and the ability to save a couple of fm stations, but this is a very WIP proyect.

About you test, is all fine with band names, appears all names, any bug?

Also feel free to make any questions and suggestions I will help you when I can also if you use Facebook you would join the ats 20 users group

73 de CA2PEY patricio Chile.
